server/tserver/src/main/java/org/apache/accumulo/tserver/TabletServer.java

```diff
@@ -948,6 +948,11 @@ public long startUpdate(TInfo tinfo, TCredentials credentials, TDurability tdura
     }
 
     private void setUpdateTablet(UpdateSession us, KeyExtent keyExtent) {
+
+      int activeThreadCount = Thread.activeCount();
```

Comment:
A few general questions about using Thread.activeCount() for this situation.  

What thread group would this measure?  From a quick look, I do not see where the calling thread is launched from a specific thread group.  Is this going to return the active threads for the tserver in general? If that would be the case, how would you recommend a "good" range for setting the limit?  The number of active threads is a tserver seems like it would be highly variable and complex to reason about and may not be a suitable surrogate for measuring the number of active writers. I don't believe that Thread.activeCount() is equivalent to the ThreadPoolExecutor getActiveCount method, and getActiveCount() requires that these writing threads submitted to run in a pool.

Assuming that Thread.activeCount is returning the number of active tserver threads, there is the potential for a lot of concurrent activity in a tserver, scans, bulk imports,... - would a "busy" tserver begin to defer writing mutations even if there were a limited number of writers active?   
 
Second, the javadoc states that 

> This method is intended primarily for debugging and monitoring purposes

Does that suggest that it may not be the best choice for run-time decisions?

Reply:
If we're going implement a thread limit for any feature, we should do so with an explicit thread pool, not by monitoring the total number of threads in the process.

Reply:
Ok, that makes sense. Could you elaborate on how I would do so?